openrelik_worker_common/mount_utils.py

```diff
@@ -0,0 +1,266 @@
+        if partition["size"] < 100000000:
+            return False
+        fs_type = self._get_fstype(f"/dev/{partition["name"]}")
+        if fs_type not in [
+            "dos",
+            "xfs",
+            "ext2",
+            "ext3",
+            "ext4",
+            "ntfs",
+            "vfat",
+        ]:
+            return False
+
+        return True
+
+    def _get_fstype(self, devname: str):
+        """Analyses the file system type of a block device or partition.
+
+        Args:
+            dev_name (str): block device or patitions device name.
+
+        Returns:
+            str: The filesystem type.
+        """
+        blkid_command = ["sudo", "blkid", "-s", "TYPE", "-o", "value", f"{devname}"]
+
+        process = subprocess.run(
+            blkid_command, capture_output=True, check=False, text=True
+        )
+        if process.returncode == 0:
+            return process.stdout.strip()
+        else:
+            raise RuntimeError(
+                f"Error running blkid on {devname}: {process.stderr} {process.stdout}"
+            )
+
+        return None
+
+    def mount(self, partition_name: str = ""):
+        """Mounts a disk or one or more partititions on a mountpoint.
+
+        Args:
+            partitions_name (str): Nameof specific partition to mount.
+
+        Returns:
+            list: A list of paths the disk/partitions have been mounted on.
+        """
+        to_mount = []
+
+        if partition_name and partition_name not in self.partitions:
+            raise RuntimeError(
+                f"Error running mount: partition name {partition_name} not found"
+            )
+
+        if partition_name:
+            to_mount.append(partition_name)
+        elif not self.partitions:
+            to_mount.append(self.blkdevice)
+        elif self.partitions:
+            to_mount = self.partitions
+
+        if not to_mount:
+            raise RuntimeError(f"Error: nothing to mount")
+
+        for mounttarget in to_mount:
+            logger.info(f"Trying to mount {mounttarget}")
+            mount_command = ["sudo", "mount"]
+            fstype = self._get_fstype(mounttarget)
+            if fstype == "xfs":
+                mount_command.extend(["-o", "ro,norecovery"])
+            elif fstype in ["ext2", "ext3", "ext4"]:
+                mount_command.extend(["-o", "ro,noload"])
+            else:
+                mount_command.extend(["-o", "ro"])
+
+            mount_command.append(mounttarget)
+
+            mount_folder = f"{self.mountroot}/{uuid4().hex}"
+            os.makedirs(mount_folder)
+
+            mount_command.append(mount_folder)
+
+            process = subprocess.run(
+                mount_command, capture_output=True, check=False, text=True
+            )
+            if process.returncode == 0:
+                logger.info(f"Mounted {mounttarget} to {mount_folder}")
+                self.mountpoints.append(mount_folder)
+            else:
+                raise RuntimeError(
```

Comment:
Do we want to log here that we failed to mount the partition or is the caller expected to log errors if they catch RuntimeError? Same comment in other places were RuntimeError is raised.